### PR TITLE
[AMDGPU] Remove dead pass: AMDGPUMachineCFGStructurizer

### PR DESCRIPTION
**Pull request by jmmartinez**
The two deleted functions in `SIInstrInfo` where only referenced in the deleted pass.

```cpp
  void convertNonUniformIfRegion(MachineBasicBlock *IfEntry,
                                 MachineBasicBlock *IfEnd) const;

  void convertNonUniformLoopRegion(MachineBasicBlock *LoopEntry,
                                   MachineBasicBlock *LoopEnd) const;
```

The only thing I'm not really sure about is the change in `SIInstructions.td`.

### CODE REVIEW

File: llvm/lib/Target/AMDGPU/SIInstructions.td [hunk 1]
```diff
@@ -421,13 +421,11 @@ def IGLP_OPT : SPseudoInstSI<(outs), (ins i32imm:$mask),
...
 let isTerminator = 1, isNotDuplicable = 1 in {
 
-let OtherPredicates = [EnableLateCFGStructurize] in {
- def SI_NON_UNIFORM_BRCOND_PSEUDO : CFPseudoInstSI <
-  (outs),
-  (ins SReg_1:$vcc, brtarget:$target),
-  [(brcond i1:$vcc, bb:$target)]> {
-    let Size = 12;
-}
+def SI_NON_UNIFORM_BRCOND_PSEUDO : CFPseudoInstSI <
```
Comment (arsenm) [inline llvm/lib/Target/AMDGPU/SIInstructions.td:424]: Should be able to delete this and any of the references too

File: llvm/lib/Target/AMDGPU/SIInstrInfo.cpp [hunk 2]
```diff
@@ -8776,75 +8776,6 @@ bool SIInstrInfo::isNonUniformBranchInstr(MachineInstr &Branch) const {
   return Branch.getOpcode() == AMDGPU::SI_NON_UNIFORM_BRCOND_PSEUDO;
 }
```
Comment (jayfoad) [inline llvm/lib/Target/AMDGPU/SIInstrInfo.cpp:8777]: This function `isNonUniformBranchInstr` is also unused now.

**Comment by arsenm** [comment]
As a follow up we should also be able to delete DisableStructurizer

**Comment by jmmartinez** [comment]
I've created the follow up merge-request in https://github.com/llvm/llvm-project/pull/105645

**Review by jayfoad — APPROVED** [review]
LGTM

**Pull request merged**